crates/chain/tests/test_keychain_txout_index.rs

```diff
@@ -44,29 +52,133 @@ fn spk_at_index(descriptor: &Descriptor<DescriptorPublicKey>, index: u32) -> Scr
         .script_pubkey()
 }
 
+#[test]
+fn append_keychain_derivation_indices() {
```

Comment:
Nit: I found it hard to figure out what this test did based on the name. What do you think about this?

```suggestion
fn append_changesets_check_last_revealed() {
```

Also, I think it will help if we had a short doc-comment explaining the test as well.